Framework/PythonInterface/mantid/api/src/Exports/ExperimentInfo.cpp

```diff
@@ -77,12 +78,12 @@ void export_ExperimentInfo() {
       .staticmethod("getInstrumentFilename")
 
       .def("sample", &ExperimentInfo::sample,
-           return_value_policy<reference_existing_object>(), args("self"),
+           return_value_policy<copy_const_reference>(), args("self"),
            "Return the :class:`~mantid.api.Sample` object. This cannot be "
            "modified, use mutableSample to modify.")
 
       .def("mutableSample", &ExperimentInfo::mutableSample,
-           return_value_policy<reference_existing_object>(), args("self"),
+           return_value_policy<copy_non_const_reference>(), args("self"),
```

Comment:
I think I slightly misunderstood the intent in our other discussion. I was thinking we would just remove the `no_init` and `boost::noncopyable` from `Sample`, `Run` etc. For `mutableSample` I think we have to keep the return by reference or there will be no way to change anything in the sample attached to a workspace from Python.

We can copy the sample for the `sample` method but is there a performance concern if I want to access some information on and then it does a full copy each time?

Reply:
@martyngigg yes, with `copy_const_reference` you would indeed invoke a copy every time. Is the performance concern significant enough that this is not a good approach given the current use-cases?

Reply:
I think the sample could be heavy enough to copy that it would be a concern.